woocommerce/woocommerce
# Add support to install master branch when using WooCommerce Beta Tester

## Changes

* Fix a bug where the admin menu wasn't being registered early enough, causing it to not be added to the navigatable pages
* Add support for installing the trunk branch in the UI and add a little more error handling for failing to fetch manifest
* Add support to WP CLI for installing the trunk branch

## Testing Instructions

1. Checkout this branch, mount the beta tester plugin in your local dev environment. Also have WooCommerce installed and activated.
2. Ensure composer has been run (it happens on pnpm postinstall)
3. Ensure JS assets are built. (In dev fastest way is just run `pnpm start` from the beta tester directory)
4. Now if you navigate to the live branches screen in wp-admin via `WooCommerce > Live Branches` you should see that trunk is listed as an available branch to install in the dropdown. Try "Install and Activate". It should be installed and activated as the current version
5. Now ideally reset your environment to test WP-CLI (let me know if you need assistance with using wp-cli in your env, its different for every env setup).  Run `wp wc-beta-tester install trunk` and then `wp wc-beta-tester deactivate_woocommerce` and then `wp wc-beta-tester activate trunk`. These commands should all succeed then you can go to the live branches menu in wp-admin as in step 4, and see that trunk is installed and activated.
